vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -448,10 +448,17 @@ and
                                                 optionsAssociation.Remove(projectContext) |> ignore
                                                 project.Disconnect()))
                 for referencedSite in ProjectSitesAndFiles.GetReferencedProjectSites (site, this.SystemServiceProvider) do
-                    let referencedProjectId = setup referencedSite                    
+                    let referencedProjectFileName = referencedSite.ProjectFileName()
+                    let referencedProjectDisplayName = projectDisplayNameOf referencedProjectFileName
+                    let referencedProjectId = workspace.ProjectTracker.GetOrCreateProjectIdForPath(referencedProjectFileName, referencedProjectDisplayName)
                     project.AddProjectReference(ProjectReference referencedProjectId)
-                workspace.ProjectTracker.AddProject(project)
-            projectId
+
+                if not (workspace.ProjectTracker.ContainsProject(project)) then 
+                    workspace.ProjectTracker.AddProject(project)
+
+                for referencedSite in ProjectSitesAndFiles.GetReferencedProjectSites (site, this.SystemServiceProvider) do
```

Comment:
@Pilchie Currently we do it in this order:

                let project = projectContextFactory.CreateProjectContext(...)
                let referencedProjectId1 = workspace.ProjectTracker.GetOrCreateProjectIdForPath(...)
                project.AddProjectReference(ProjectReference referencedProjectId1)
                let referencedProjectId2 = workspace.ProjectTracker.GetOrCreateProjectIdForPath(...)
                project.AddProjectReference(ProjectReference referencedProjectId2)
                workspace.ProjectTracker.AddProject(project)
                ... then setup referencedSite1 ....
                ... then setup referencedSite 2....

Is that right?  Thanks